lib/node_modules/@stdlib/blas/ext/index-of/lib/main.js

```diff
@@ -0,0 +1,204 @@
+
+
+// MAIN //
+
+/**
+* Returns the first index of a specified search element along one or more ndarray dimensions.
+*
+* @name indexOf
+* @type {Function}
+* @param {ndarrayLike} x - input ndarray
+* @param {(ndarrayLike|number)} searchElement - search element
+* @param {(ndarrayLike|number)} [fromIndex] - index from which to begin searching
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform operation
+* @param {boolean} [options.keepdims=false] - boolean indicating whether the reduced dimensions should be included in the returned ndarray as singleton dimensions
+* @param {string} [options.dtype] - output ndarray data type
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} search element argument must be either an ndarray-like object or a numeric value
+* @throws {TypeError} from index argument must be either an ndarray-like object or a numeric value
+* @throws {TypeError} options argument must be an object
+* @throws {RangeError} dimension indices must not exceed input ndarray bounds
+* @throws {RangeError} number of dimension indices must not exceed the number of input ndarray dimensions
+* @throws {Error} must provide valid options
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, -3.0, 4.0, -5.0, 6.0 ] );
+*
+* // Define the shape of the input array:
+* var sh = [ 3, 1, 2 ];
+*
+* // Define the array strides:
+* var sx = [ 2, 2, 1 ];
+*
+* // Define the index offset:
+* var ox = 0;
+*
+* // Create an input ndarray:
+* var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
+*
+* // Find index:
+* var out = indexOf( x, 4.0 );
+* // returns <ndarray>
+*
+* var idx = out.get();
+* // returns 3
+*/
+function indexOf( x, searchElement, fromIndex, options ) {
+	var nargs;
+	var opts;
+	var ord;
+	var dt;
+	var sh;
+
+	nargs = arguments.length;
+
+	// Resolve input ndarray meta data:
+	dt = getDType( x );
+	ord = getOrder( x );
+
+	// Case: indexOf( x, searchElement )
+	if ( nargs === 2 ) {
+		// Case: indexOf( x, searchElement_ndarray )
+		if ( isndarrayLike( searchElement ) ) {
+			// As the operation is performed across all dimensions, `searchElement` is assumed to be a zero-dimensional ndarray...
+			return base( x, searchElement, broadcastScalar( 0, DEFAULT_DTYPE, [], ord ) ); // eslint-disable-line max-len
+		}
+		// Case: indexOf( x, searchElement_scalar )
+		if ( isNumber( searchElement ) ) {
+			return base( x, broadcastScalar( searchElement, dt, [], ord ), broadcastScalar( 0, DEFAULT_DTYPE, [], ord ) ); // eslint-disable-line max-len
+		}
+		throw new TypeError( format( 'invalid argument. Second argument must be either an ndarray or a scalar value. Value: `%s`.', searchElement ) );
```

Comment:
This error should not be possible. When `searchElement` is not an ndarray-like object, it should always be considered a "scalar".